// react-carousel-mui/dist/components/Carousel.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemsPerPage = void 0;
var React = __importStar(require("react"));
var ChevronLeft_1 = __importDefault(require("@mui/icons-material/ChevronLeft"));
var ChevronRight_1 = __importDefault(require("@mui/icons-material/ChevronRight"));
var useSlider_1 = __importDefault(require("../hooks/useSlider"));
var CommonButton_1 = __importDefault(require("./CommonButton"));
var Wrapper_1 = __importDefault(require("./Wrapper"));
var hooks_1 = require("../hooks");
var ItemsPerPage;
(function (ItemsPerPage) {
    ItemsPerPage["xs"] = "xs";
    ItemsPerPage["sm"] = "sm";
    ItemsPerPage["md"] = "md";
    ItemsPerPage["lg"] = "lg";
    ItemsPerPage["xl"] = "xl";
    ItemsPerPage["tablet"] = "tablet";
})(ItemsPerPage = exports.ItemsPerPage || (exports.ItemsPerPage = {}));
var Carousel = function (_a) {
    var items = _a.items, _b = _a.itemGap, itemGap = _b === void 0 ? 16 : _b, itemsPerPage = _a.itemsPerPage, itemRenderer = _a.itemRenderer, customPrevButton = _a.customPrevButton, customNextButton = _a.customNextButton, other = __rest(_a, ["items", "itemGap", "itemsPerPage", "itemRenderer", "customPrevButton", "customNextButton"]);
    var breakPoint = (0, hooks_1.useBreakpoint)();
    var _c = (0, useSlider_1.default)(__assign({ itemGap: itemGap, itemsPerPage: itemsPerPage }, other)), currentItemIndex = _c.currentItemIndex, itemWidth = _c.itemWidth, maxWidth = _c.maxWidth, switchSlide = _c.switchSlide;
    var isMobile = breakPoint === 'xs';
    var ipp = itemsPerPage[breakPoint];
    var shouldShowNext = currentItemIndex < items.length - ipp;
    var shouldShowPrev = currentItemIndex > 0;
    return (React.createElement(Wrapper_1.default, { sx: {
            position: 'relative',
            maxWidth: maxWidth,
        } },
        shouldShowPrev
            && (customPrevButton || (React.createElement(CommonButton_1.default, { sxButton: { left: -17 }, onClick: switchSlide('prev') },
                React.createElement(ChevronLeft_1.default, null)))),
        React.createElement(Wrapper_1.default, { sx: {
                overflow: isMobile ? 'auto' : 'hidden',
                maxWidth: '100%',
                msOverflowStyle: 'none',
                scrollbarWidth: 'none',
                '&::-webkit-scrollbar': {
                    display: 'none',
                },
            } },
            React.createElement(Wrapper_1.default, { sx: {
                    width: items.length * itemWidth + (items.length - 1) * itemGap,
                    display: 'grid',
                    gridAutoFlow: 'column',
                    gridColumnGap: itemGap,
                    position: 'relative',
                    left: "-".concat(currentItemIndex * itemWidth + currentItemIndex * itemGap, "px"),
                    transition: 'left 200ms ease-in-out',
                } }, items.map(function (slide) { return React.cloneElement(itemRenderer(slide), {
                style: { width: itemWidth },
            }); }))),
        shouldShowNext
            && (customNextButton || (React.createElement(CommonButton_1.default, { sxButton: { right: -17 }, onClick: switchSlide('next') },
                React.createElement(ChevronRight_1.default, null))))));
};
exports.default = Carousel;
